src/components/src/ai-assistant/ai-assistant-functions.tsx

```diff
@@ -0,0 +1,137 @@
+import { ActionHandler } from '@kepler.gl/actions';
+import { mapStyleChange } from '@kepler.gl/actions';
+import { DEFAULT_MAP_STYLES } from '@kepler.gl/constants';
+import {
+  CallbackFunctionProps,
+  CustomFunctionCall,
+  CustomFunctionContext,
+  CustomFunctionOutputProps,
+  ErrorCallbackResult,
+  RegisterFunctionCallingProps
+} from 'ai-assistant';
+import React, { ReactNode, useEffect, useState } from 'react';
+import MapStyleSelectorFactory from '../side-panel/map-style-panel/map-style-selector';
+import { appInjector } from '../container';
+import { MapStyle } from '@kepler.gl/reducers';
+
+export function basemapFunctionDefinition(
```

Comment:
is this the actual integration? I am not sure we should have that mixes with React and state management... it would be good to have clear separation.